src/mongo/db/query/get_executor.cpp

```diff
@@ -306,7 +306,20 @@ void fillOutPlannerParams(OperationContext* opCtx,
         auto collDesc = CollectionShardingState::get(opCtx, canonicalQuery->nss())
                             ->getCollectionDescription(opCtx);
         if (collDesc.isSharded()) {
-            plannerParams->shardKey = collDesc.getKeyPattern();
+            const auto keyPattern = collDesc.getKeyPattern();
+            ShardKeyPattern shardKeyPattern(keyPattern);
+
+            // If we're specifying one shard exactly, we're always going to land on exactly one
```

Comment:
I find the first sentence of this comment a little awkward. I think the intention was to say "if we're specifying an exact value for the shard key, we're always going ...". I would suggest rewording the comment to something like this:

```
// If the shard key is specified exactly, the query is guaranteed to only target one
// shard. Shards cannot own orphans for the key ranges they own, so there is no need
// to include a shard filtering stage. By omitting the shard filter, it may be possible
// to get a more efficient plan (for example, a COUNT_SCAN may be used if the query is
// eligible).
```